### PR TITLE
Improve responsiveness

### LINKED ISSUE #4: Right sidebar & notifications panel

### LINKED ISSUE #13: Refactor navbar. The current structure is not good
The current structure is had to understand and has duplicated code. This should be improved.

### LINKED ISSUE #19: Improve responsiveness
* Remove header on low-res.
* Make right-sidebar full-screen on low-res.
* Add a close-button to right-sidebar in low-res.
* Make sure that I have a common breakpoint somewhere that determines when cellphone styles kicks in.

### LINKED ISSUE #25: improve appearance of navbar transition

### PR DESCRIPTION
**Pull request by fredrik-hjarner**
A lot of changes!

Primarily improved responsiveness.

* Split Navbar into several components:
  * Removed the Pusher/Pushable from semantic-ui-react.
  * _navbar-layout.js_ is the root file
  * _navbar.js_ contains **only** the navbar itself
  * top-level-menu and subcomponents are under _/navbar-components/_
  * Removed SmallNavbar. Now there's only one navbar that animates to be big/small.
* RightSidebar and Notifications are responsive.

closes #19 
closes #4 
closes #13
closes #25

### CODE REVIEW

File: src/components/navbar/navbar.js [hunk 1]
```diff
@@ -1,116 +1,29 @@
...
-    }
-    return (
-      <Menu.Menu key={index}>
-        {textArray.map(this.createSubMenuLink)}
-      </Menu.Menu>
-    );
-  }
+export const Navbar = (props) => {
+  const { visible, sideMenuContent, activeItem, setActiveItem } = props;
+  const navbarStyles = `no-margins-or-padding min-height-100${visible ? ' navbar' : ' small-navbar'}`;
```
Comment (simborg) [inline src/components/navbar/navbar.js]: add the space before the $ instead of in the extra classes

File: src/components/sticky/sticky.js [hunk 2]
```diff
@@ -43,13 +42,13 @@ export class Sticky extends Component<Props, State> {
...
-    const rightPos = navbarIsBig ? 'navbar-is-big' : 'navbar-is-small';
 
     return (
-      <div styleName={`${moving} ${rightPos}`}>
-        {this.props.children}
+      <div id='sticky-root-element'>
+        <div styleName={moving}>
+          {this.props.children}
+        </div>
       </div>
```
Comment (simborg) [inline src/components/sticky/sticky.js]: is the sticky still needed? didn't 'fixed' on right-sidebar adjust this?

Comment (fredrik-hjarner) [reply src/components/sticky/sticky.js]: fixat

File: src/components/forms/change-password-form/change-password-form.js [hunk 3]
```diff
@@ -40,10 +40,15 @@ const validate = ({ oldPassword, newPassword, confirmPassword }) => {
 };
 
 const ChangePasswordReduxForm = (props) => {
-  const { submitting, onSubmit, handleSubmit, error } = props;
+  const { submitting, onSubmit, handleSubmit, error, reset } = props;
 
   return (
-    <Form onSubmit={handleSubmit(onSubmit)}>
+    <Form
+      onSubmit={handleSubmit((data) => {
```
Comment (Valmorbus) [inline src/components/forms/change-password-form/change-password-form.js]: bind submit to a function instead

File: src/components/forms/change-password-form/change-password-form.js [hunk 4]
```diff
@@ -40,10 +40,15 @@ const validate = ({ oldPassword, newPassword, confirmPassword }) => {
...
+        reset();
+        return onSubmit(data);
```
Comment (simborg) [inline src/components/forms/change-password-form/change-password-form.js]: bind a function as submit instead.
why are you returning the submit function?

File: src/components/header/style/header.scss [hunk 5]
```diff
@@ -2,34 +2,32 @@
...
 }
 
-.menu-button {
-
-}
-
+/*
+ * For the navbar toggle button that comes
+ * up after scrolling down.
+ */
```
Comment (simborg) [inline src/components/header/style/header.scss]: is this comment necessary? cant the classname be used to explain functionality?

File: src/components/navbar/navbar-container.js [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+import { Toastrs } from 'components/toastrs/toastrs';
+import { sideMenuContent } from './side-menu-content';
+import { Navbar } from './navbar';
+import './style/navbar.scss';
+
+export class NavbarContainer extends Component {
+  static propTypes = {
+    children: PropTypes.object.isRequired,
+    history: PropTypes.object.isRequired,
+    //                   the eslint-disable-line below is needed
```
Comment (simborg) [inline src/components/navbar/navbar-container.js]: remove comment

File: src/routes/dashboard/dashboard.js [hunk 7]
```diff
@@ -1,10 +1,196 @@
...
+  //     <Fragment>
+  //       <Button color='green' floated='right'>
+  //         From
+  //       </Button>
+  //       <Button color='green' floated='right'>
+  //         To
+  //       </Button>
+  //     </Fragment>
+  //   );
+  // }
```
Comment (simborg) [inline src/routes/dashboard/dashboard.js]: remove

File: src/components/right-sidebar/right-sidebar.js [hunk 8]
```diff
@@ -15,11 +15,15 @@ import { push } from 'react-router-redux';
 import { logout } from 'redux/modules/user';
 import { openToastr } from 'redux/modules/toastr';
 import { ChangePasswordForm } from 'components/forms';
-import { toastrType } from 'components/toastr/toastr';
+import { toastrType } from 'components/toastrs/toastr/toastr';
```
Comment (Valmorbus) [inline src/components/right-sidebar/right-sidebar.js]: cant find these here? did you add them?

File: src/components/toastrs/toastr/toastr.js [hunk 9]
```diff
@@ -79,7 +79,7 @@ class Toastr extends React.Component {
         message = (
           <Message
             success
-            styleName='toastr'
+            styleName='success-toastr'
             icon='checkmark box'
             header={header}
             content={content}
```
Comment (simborg) [inline src/components/toastrs/toastr/toastr.js]: cant these be a function in the toastr.js file instead? seems like to much separation

File: src/routes/dashboard/dashboard.js (hunk 7 above)
Comment (fredrik-hjarner) [reply src/routes/dashboard/dashboard.js]: ah good find. I must have missed that part.

File: src/components/forms/change-password-form/change-password-form.js (hunk 4 above)
Comment (fredrik-hjarner) [reply src/components/forms/change-password-form/change-password-form.js]: If I get it correctly...

Normally one would do:
```
onSubmit={handleSubmit(onSubmit)}
```

handleSubmit(onSubmit) returns a function that takes the event as an argument. It then sends the event to the onSubmit.
In other words handleSubmit(onSubmit)(event) can be written as:
```
handleSubmit(event => onSubmit(event))
```

which can be written as:
```
handleSubmit(event => {
  return onSubmit(event);
})
```

The code below is my way of doing the exact same thing but inject a execution of this.props.reset():

```
handleSubmit((event) => {
  reset();
  return onSubmit(event);
}
```

**Pull request merged**